pkg/oc/cli/newapp/newapp.go

```diff
@@ -1019,7 +1019,11 @@ func TransformRunError(err error, baseName, commandName, commandPath string, gro
 		// TODO: suggest things to the user
 		groups.Add("", "", "", UsageError(commandPath, newAppNoInput, baseName, commandName))
 	default:
-		groups.Add("", "", "", err)
+		if runtime.IsNotRegisteredError(err) {
+			groups.Add("", "", "", fmt.Errorf("The template contained an object type not registered with the runtime decoder used by `oc new-app`.  Use `oc process | oc create` instead.  The decoder returned: %s", err.Error()))
```

Comment:
I'd simplify the message a bit further - "the template contained an unknown object type - use `oc process | oc create` instead". No need to report the underlying error since we know why we failed.

Reply:
I like the text simplification.  Will push that update in a bit.  But beforehand,
I'm wary of not reporting the underlying error.  I find some indication of the offending kind will be useful.  Unfortunately, the not registered error does not publicly expose the means of getting just the kind, but I think it is worth the additional characters.

Swayed?

Reply:
Good point on reporting the offending kind - chances are there are few of those in a template. I'm swayed.